### PR TITLE
OSDOCS10256: Added a section regarding FIPS compliant support for cert manager

### PR DESCRIPTION
**Pull request by subhtk**
Version(s): 4.14+

Issue: https://issues.redhat.com/browse/OSDOCS-10256

Link to docs preview: https://76688--ocpdocs-pr.netlify.app/openshift-enterprise/latest/security/cert_manager_operator/index.html#cert-manager-fips-support_cert-manager-operator-about

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/cert-manager-fips-procedure.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* Enable the use of cgo using `CGO_ENABLED=1`.
+* Ensure OpenSSL library is included.
+* Enforce strict FIPS mode at runtime.
+* Ensure no conflicting `OpenSSL` version.
+* Verify binaries are dynamically linked.
+
+. Add `check-payload` tool to your CI workflows to automate FIPS compliance checks.
+
+. Configure your CI environment to run with FIPS enabled.
```
Comment (swghosh) [inline modules/cert-manager-fips-procedure.adoc:43]: As discussed, we can consider removing this section about the check-payload tool. Essentially, that'd strip out the Procedure section IMO.

File: security/cert_manager_operator/index.adoc [hunk 2]
```diff
@@ -20,9 +20,15 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
+
+include::modules/cert-manager-fips-procedure.adoc[leveloffset=+2]
 
 [role="_additional-resources"]
 [id="cert-manager-operator-about_additional-resources"]
 == Additional resources
 
 * link:https://cert-manager.io/docs/[cert-manager project documentation]
+
+* link:https://access.redhat.com/articles/5059881[OpenShift security guide book]
```
Comment (swghosh) [inline security/cert_manager_operator/index.adoc:34]: As discussed, consider double-checking if the security guide book is publicly accessible and if yes, if it's a good fit to share with customers/partners externally for their reference.

File: security/cert_manager_operator/index.adoc [hunk 3]
```diff
@@ -20,9 +20,15 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
+//FIPS compliant support
+include::modules/cert-manager-fips-support.adoc[leveloffset=+1]
+
+include::modules/cert-manager-fips-procedure.adoc[leveloffset=+2]
 
 [role="_additional-resources"]
 [id="cert-manager-operator-about_additional-resources"]
 == Additional resources
 
 * link:https://cert-manager.io/docs/[cert-manager project documentation]
```
Comment (swghosh) [inline security/cert_manager_operator/index.adoc:32]: Also, you can consider including the following FIPS related pages:
- https://docs.openshift.com/container-platform/4.14/security/container_security/security-compliance.html
- https://docs.openshift.com/container-platform/4.14/installing/installing-fips.html#installing-fips-mode_installing-fips

File: modules/cert-manager-fips-support.adoc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/index.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="cert-manager-fips-support_{context}"]
+= About FIPS compliance for {cert-manager-operator}
+
+For many {product-title} customers, regulatory readiness, or compliance, on some level is required before any systems can be put into production. That regulatory readiness can be imposed by national standards, industry standards or the organization's corporate governance framework. The {cert-manager-operator} now supports Federal Information Processing Standards (FIPS) compliance, a crucial requirement for highly secure environments to ensure the use of approved cryptographic technologies within the cluster.
```
Comment (swghosh) [inline modules/cert-manager-fips-support.adoc:9]: Looks good to me, except maybe you can add "FIPS 140" added specifically.
Also, you can add that with the new changes cert-manager now uses NIST validated crypto libraries.

Comment (swghosh) [reply modules/cert-manager-fips-support.adoc:9]: Additionally, I found a slide deck with some FIPS wordings
maybe you could refer that as well:  [Red Hat Internal] https://docs.google.com/presentation/d/1Rt0AoUK12__2CQgWM_Qk9A7GPEqKMO4mQZhW3mA4pU8/edit#slide=id.g2696a798143_1_0
since we're (cert-manager-operator) technically an OpenShift Layered product.

File: modules/cert-manager-fips-procedure.adoc (hunk 1 above)
Comment (subhtk) [reply modules/cert-manager-fips-procedure.adoc:43]: Yes. as discussed, I will remove the section.

File: modules/cert-manager-fips-support.adoc (hunk 4 above)
Comment (subhtk) [reply modules/cert-manager-fips-support.adoc:9]: Great. Let me rephrase it.

File: modules/cert-manager-fips-support.adoc [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/index.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="cert-manager-fips-support_{context}"]
+= About FIPS compliance for {cert-manager-operator}
+
+Red Hat {cert-manager-operator} is designed for FIPS. When running on {product-title} in FIPS mode, it uses the RHEL cryptographic libraries submitted to NIST for FIPS Validation on only the x86_64, ppc64le, and s390X architectures. For more information about the NIST validation program, see "cryptographic module validation program". For the latest NIST status for the individual versions of the RHEL cryptographic libraries submitted for validation, see "compliance activities and government standards".
```
Comment (swghosh) [inline modules/cert-manager-fips-support.adoc:9]: Possible repetition of the word "Red Hat". Also, we need to include that FIPS support is only available in 1.14.0+.
```suggestion
{cert-manager-operator} is designed for FIPS starting v1.14.0 and above. When running on {product-title} in FIPS mode, it uses the RHEL cryptographic libraries submitted to NIST for FIPS Validation on only the x86_64, ppc64le, and s390X architectures. For more information about the NIST validation program, see "cryptographic module validation program". For the latest NIST status for the individual versions of the RHEL cryptographic libraries submitted for validation, see "compliance activities and government standards". 
```

**Review by swghosh — APPROVED** [review]
Minor nit, LGTM otherwise.

**Comment by lihongan** [comment]
@lunarwhite could you please help take a look? thanks.
cc @xingxingxia

File: security/cert_manager_operator/index.adoc [hunk 6]
```diff
@@ -20,9 +20,16 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
 
+//FIPS compliant support
+include::modules/cert-manager-fips-support.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 [id="cert-manager-operator-about_additional-resources"]
 == Additional resources
 
 * link:https://cert-manager.io/docs/[cert-manager project documentation]
+* xref:../../security/container_security/security-compliance.adoc#security-compliance[Configuring an ACME issuer]
```
Comment (lunarwhite) [inline security/cert_manager_operator/index.adoc:31]:
```suggestion
* xref:../../security/container_security/security-compliance.adoc[Understanding compliance]
```
It should be a typo

File: security/cert_manager_operator/index.adoc [hunk 7]
```diff
@@ -20,9 +20,16 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
+* xref:../../installing/installing-fips.adoc#installing-fips-mode_installing-fips[Installing a cluster in FIPS mode]
+* link:https://csrc.nist.gov/Projects/cryptographic-module-validation-program/validated-modules[Cryptographic module validation program]
+* link:https://access.redhat.com/articles/2918071#fips-140-2-and-fips-140-3-2[Compliance activities and government standards]
+* link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html-single/installing/index#installing-preparing-security[2.1.4. Do you need extra security for your cluster?]
```
Comment (lunarwhite) [inline security/cert_manager_operator/index.adoc:35]: I'm wondering if we could move these three references under "Additional Resources" to the main text as hyperlinks correspondingly.

Comment (subhtk) [reply security/cert_manager_operator/index.adoc:35]: Apologies, I cannot add xrefs in modules. Hyperlinks can be added in assembly.

File: modules/cert-manager-fips-support.adoc [hunk 8]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/index.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="cert-manager-fips-support_{context}"]
+= About FIPS compliance for {cert-manager-operator}
+
+{cert-manager-operator} is designed for FIPS starting v1.14.0 and above. When running on {product-title} in FIPS mode, it uses the RHEL cryptographic libraries submitted to NIST for FIPS Validation on only the x86_64, ppc64le, and s390X architectures. For more information about the NIST validation program, see "cryptographic module validation program". For the latest NIST status for the individual versions of the RHEL cryptographic libraries submitted for validation, see "compliance activities and government standards".
```
Comment (jeana-redhat) [inline modules/cert-manager-fips-support.adoc:9]:
```suggestion
Starting with version 1.14.0, {cert-manager-operator} is designed for FIPS compliance. When running on {product-title} in FIPS mode, it uses the RHEL cryptographic libraries submitted to NIST for FIPS validation on the x86_64, ppc64le, and s390X architectures. For more information about the NIST validation program, see "cryptographic module validation program". For the latest NIST status for the individual versions of the RHEL cryptographic libraries submitted for validation, see "compliance activities and government standards".
```

Another thing to consider here: both the supplementary resources are external links rather than xrefs, so you could stick them right in the text for easier findability. So: 

```suggestion
Starting with version 1.14.0, {cert-manager-operator} is designed for FIPS compliance. When running on {product-title} in FIPS mode, it uses the RHEL cryptographic libraries submitted to NIST for FIPS validation on the x86_64, ppc64le, and s390X architectures. For more information about the NIST validation program, see link:https://csrc.nist.gov/Projects/cryptographic-module-validation-program/validated-modules[Cryptographic module validation program]. For the latest NIST status for the individual versions of the RHEL cryptographic libraries submitted for validation, see link:https://access.redhat.com/articles/2918071#fips-140-2-and-fips-140-3-2[Compliance activities and government standards].
```

File: modules/cert-manager-fips-support.adoc [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+To enable FIPS mode, you must install {cert-manager-operator} on an Openshift Cluster configured to operate in FIPS mode. For more information, see "2.1.4. Do you need extra security for your cluster?”
```
Comment (jeana-redhat) [inline modules/cert-manager-fips-support.adoc:11]: The "2.1.4" heading number is not part of the heading and can change w/o notice:
```suggestion
To enable FIPS mode, you must install {cert-manager-operator} on an {product-title} cluster configured to operate in FIPS mode. For more information, see "Do you need extra security for your cluster?"
```

File: security/cert_manager_operator/index.adoc [hunk 10]
```diff
@@ -20,9 +20,16 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
 
 [role="_additional-resources"]
 [id="cert-manager-operator-about_additional-resources"]
 == Additional resources
 
 * link:https://cert-manager.io/docs/[cert-manager project documentation]
+* xref:../../security/container_security/security-compliance.adoc#security-compliance[Understanding compliance]
+* xref:../../installing/installing-fips.adoc#installing-fips-mode_installing-fips[Installing a cluster in FIPS mode]
+* link:https://csrc.nist.gov/Projects/cryptographic-module-validation-program/validated-modules[Cryptographic module validation program]
+* link:https://access.redhat.com/articles/2918071#fips-140-2-and-fips-140-3-2[Compliance activities and government standards]
```
Comment (jeana-redhat) [inline security/cert_manager_operator/index.adoc:34]: These two could be moved up into the text, as they are external links rather than xrefs

File: security/cert_manager_operator/index.adoc [hunk 11]
```diff
@@ -20,9 +20,16 @@ include::modules/cert-manager-request-methods.adoc[leveloffset=+1]
...
+* link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html-single/installing/index#installing-preparing-security[2.1.4. Do you need extra security for your cluster?]
```
Comment (jeana-redhat) [inline security/cert_manager_operator/index.adoc:35]: Since this is a link to OCP docs, it should probably be an xref. Omit "2.1.4." as mentioned in prev comment:
```suggestion
* xref:../../installing/installing-preparing.adoc#installing-preparing-security[Do you need extra security for your cluster?]
```

**Review by jeana-redhat — COMMENTED** [review]
Just a couple small things here - looking good overall :)

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Review by jeana-redhat — APPROVED** [review]
LGTM

/remove-label merge-review-in-progress
/remove-label merge-review-needed
